packages/scaffold-ui/src/views/w3m-account-settings-view/index.ts

```diff
@@ -310,8 +315,17 @@ export class W3mAccountSettingsView extends LitElement {
   private async onDisconnect() {
     try {
       this.disconnecting = true
-      await ConnectionController.disconnect()
-      ModalController.close()
+      const namespace = this.network?.chainNamespace as ChainNamespace
+      const connectionsByNamespace = ConnectionController.state.connections.get(namespace) ?? []
+      const hasConnections = connectionsByNamespace.length > 0
+      const connectorId = namespace && ConnectorController.state.activeConnectorIds[namespace]
+      const isMultiWalletEnabled = this.remoteFeatures?.multiWallet
+      await ConnectionController.disconnect(isMultiWalletEnabled ? { id: connectorId } : {})
+      if (hasConnections && isMultiWalletEnabled) {
+        RouterController.reset('Account')
+        RouterController.push('ProfileWallets')
```

Comment:
Same here, could it be?
```suggestion
        RouterController.replace('ProfileWallets')
```